internal/attestation/measurements/measurements.go

```diff
@@ -53,8 +53,14 @@ func (m *M) FetchAndVerify(ctx context.Context, client *http.Client, measurement
 	if err := sigstore.VerifySignature(measurements, signature, publicKey); err != nil {
 		return "", err
 	}
-	if err := yaml.Unmarshal(measurements, m); err != nil {
-		return "", err
+
+	if err := json.Unmarshal(measurements, m); err != nil {
+		if yamlErr := yaml.Unmarshal(measurements, m); yamlErr != nil {
+			return "", multierr.Append(
+				err,
+				fmt.Errorf("trying yaml format: %w", yamlErr),
+			)
+		}
```

Comment:
Can we make the legacy fall back dependent on configVersion (since we want to bump it for the next release)?

configVersion == v1 -> use YAML
configVersion == v2 -> use JSON

Reply:
No, since this code does not know about the config

Reply:
configVersion and measurements format also don't have much to do with each other, do they?

Reply:
Additional Q: Isn't the current code functionally identical to just using yaml.Unmarshal directly? json is a subset of yaml so every valid json document can be parsed by yaml.Unmarshal.

Reply:
In the later run they would _kind of_ have to do something with each other.

For newer versions (≥ v2.3.0) it would be cool if fetch-measurements would only allow the new image & measurement format, so that you cannot transparently downgrade both the image + expected measurements to try a downgrade attack. 

On the other hand, if we have backwards compatibility with v2.2.0, we would still need to accept the old format.

Having a generic fallback would skip the "better" security guarantees with the worse ones.

However, if we support configVersion v1 with v2.3.0 images, then configVersion and measurements wouldn't strictly be related with each other I guess.

At least that is my intention here. We could also add it at a later point, though.

Reply:
In the future and likely before the next release, fetch-measurements is going to need to the image UID to download the measurements anyway, which only the new configVersion will have. 

That was the intention for having a relationship between configVersion <-> measurements; eventually they are going to be related for fetch-measurements, so the idea here was that it can also work as a function toggle for YAML <-> JSON, too. Especially since fetch-measurements should later only allow measurements with a metadata header for images ≥2.3.0.

Though given that as @malt3 pointed out, yaml.Unmarshal likey can decode JSON just fine anyway so that would need to be implemented different anyway I guess.

So I assume we can just do that in an upcoming PR then.

Reply:
>Isn't the current code functionally equivalent to just using yaml.Unmarshal directly? json is a subset of yaml so every valid json document can be parsed by yaml.Unmarshal.

While in theory this would be true, in Go reality this does not work.
It might be possible if we use the Kubernetes yaml implementation instead.

>In the future and likely before the next release, fetch-measurements is going to need to the image UID to download the measurements anyway, which only the new configVersion will have.

I think you are confusing where this code section is implemented. `*m.FetchAndVerify()` is not a function of the `constellation config fetch-measurements`, but a method of `measurements.M`, which simply takes a URL and loads the measurements from there. That code section does not know, and does not need to know, the config version.

Reply:
I see where it's coming from, but not sure I agree that FetchAndVerify should actually belong to `measurements.M` vs belonging to `Config` in the first place since all actual callers are actively there to modify the config.

I don't mind how it's implemented in detail, but later on it should be possible to have metadata enforcement in some way or another.